Notebooks/DC2Cookbook.ipynb

```diff
@@ -0,0 +1,559 @@
+    "```\n",
+    "Now we translate the text file into a FITS file for indexing. I decided to change the column names from the default output by CatSim.\n",
+    "Then you can do the actual index generation.  You'll need to set up a couple of packages then run some scripts to do the formatting."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```bash\n",
+    "$> setup astrometry_net\n",
+    "$> setup pyfits\n",
+    "$> text2fits.py -H 'id, ra, dec, g, r, i, starnotgal, isvariable' -s ', ' twinkles_ref_obs.txt twinkles_ref.fits -f 'kdddddjj'\n",
+    "$> export P=0106160\n",
+    "$> build-astrometry-index -i twinkles_ref.fits -o index-${P}00.fits -I ${P}00 -P 0 -S r -n 100 -L 20 -E -j 0.4 -r 1 > build-00.log\n",
+    "$> build-astrometry-index -1 index-${P}00.fits -o index-${P}01.fits -I ${P}01 -P 1 -S r -L 20 -E -M -j 0.4 > build-01.log &\n",
+    "$> build-astrometry-index -1 index-${P}00.fits -o index-${P}02.fits -I ${P}02 -P 2 -S r -L 20 -E -M -j 0.4 > build-02.log &\n",
+    "$> build-astrometry-index -1 index-${P}00.fits -o index-${P}03.fits -I ${P}03 -P 3 -S r -L 20 -E -M -j 0.4 > build-03.log &\n",
+    "$> build-astrometry-index -1 index-${P}00.fits -o index-${P}04.fits -I ${P}04 -P 4 -S r -L 20 -E -M -j 0.4 > build-04.log\n",
+    "$> mkdir and_files\n",
+    "$> mv index*.fits and_files\n",
+    "$> cd and_files\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The matcher needs to know which index files are available and what columns to use for photometric calibration.  These are specified using a configuration file.  This file goes in the `and_files` directory.  It is called `andConfig.py` and looks like this:\n",
+    "```\n",
+    "root.starGalaxyColumn = \"starnotgal\"\n",
+    "root.variableColumn = \"isvariable\"\n",
+    "filters = ('u', 'g', 'r', 'i', 'z', 'y')\n",
+    "root.magColumnMap = {'u':'g', 'g':'g', 'r':'r', 'i':'i', 'z':'i', 'y':'i'}\n",
+    "root.indexFiles = ['index-010616000.fits',\n",
+    "'index-010616001.fits',\n",
+    "'index-010616002.fits',\n",
+    "'index-010616003.fits',\n",
+    "'index-010616004.fits']\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Set up the data to run DM processing\n",
+    "\n",
+    "First you'll need to build the stack using tickets/DM-4302 of obs_lsstSim.  In order to patch a branch version onto a pre-existing stack you can do something like the following:\n",
+    "\n",
+    "1. Build a master stack.  I suggest using [lsstsw](https://confluence.lsstcorp.org/display/LDMDG/The+LSST+Software+Build+Tool).\n",
+    "2. Set up the stack: e.g. `$> setup obs_lsstSim -t bNNNN`\n",
+    "3. Clone the package you want to patch on top of your stack `$> clone git@github.com:lsst/obs_lsstSim.git; cd obs_lsstSim`\n",
+    "4. Get the branch: `$> checkout tickets/DM-4302`\n",
```

Comment:
I wonder if we should make a DC2 branch of `obs_lsstSIm`.  We didn't do this for Twinkles or DC1 and it resulted in problems.

Reply:
I'd prefer that DESC have more direct control over the changes that would need to be made in obs_lsstSim.  I don't think that working off of branches of a repo in the lsst org gives us the control we need.  I would propose instead forking obs_lsstSim into LSSTDESC.  That will give us the freedom to control the development workflow, i.e., develop along whatever branches we want, apply release tags to master whenever we want, and thereby maintain better provenance of the code we use in our production runs.  We can always PR from our fork back to the lsst version.